db/db_impl/db_impl_compaction_flush.cc

```diff
@@ -1595,11 +1599,6 @@ Status DBImpl::CompactFilesImpl(
 
   ReleaseFileNumberFromPendingOutputs(pending_outputs_inserted_elem);
 
-  if (compaction_job_info != nullptr) {
-    BuildCompactionJobInfo(cfd, c.get(), s, compaction_job_stats,
```

Comment:
There seems to be some info only known after compaction job is run e.g, output_table_properties, compaction_job_info->status, compaction_job_info->stats. Would it be too early to call this function compaction_job.Run()?

Reply:
Good catch, updated to just release the mutex.